heudiconv/dicoms.py

```diff
@@ -0,0 +1,507 @@
+    import json
+    import re
+
+    if not min_meta:
+        import dcmstack as ds
+        stack = ds.parse_and_stack(dcmfiles, force=force).values()
+        if len(stack) > 1:
+            raise ValueError('Found multiple series')
+        stack = stack[0]
+
+        #Create the nifti image using the data array
+        if not op.exists(niftifile):
+            nifti_image = stack.to_nifti(embed_meta=True)
+            nifti_image.to_filename(niftifile)
+            return ds.NiftiWrapper(nifti_image).meta_ext.to_json()
+
+        orig_nii = nb.load(niftifile)
+        aff = orig_nii.affine
+        ornt = nb.orientations.io_orientation(aff)
+        axcodes = nb.orientations.ornt2axcodes(ornt)
+        new_nii = stack.to_nifti(voxel_order=''.join(axcodes), embed_meta=True)
+        meta = ds.NiftiWrapper(new_nii).meta_ext.to_json()
+
+    meta_info = None if min_meta else json.loads(meta)
+
+    if bids_info:
+
+        if min_meta:
+            meta_info = bids_info
+        else:
+            # make nice with python 3 - same behavior?
+            meta_info = meta_info.copy()
+            meta_info.update(bids_info)
+            # meta_info = dict(meta_info.items() + bids_info.items())
+        try:
+            meta_info['TaskName'] = (re.search('(?<=_task-)\w+',
+                                               op.basename(infofile))
+                                     .group(0).split('_')[0])
+        except AttributeError:
+            pass
+    # write to outfile
+    with open(infofile, 'wt') as fp:
+        json.dump(meta_info, fp, indent=3, sort_keys=True)
+
+    return niftifile, infofile
+
+
+def embed_metadata_from_dicoms(bids, item_dicoms, outname, outname_bids,
+                               prov_file, scaninfo, tempdirs, with_prov,
+                               min_meta):
+    """
+    Enhance sidecar information file with more information from DICOMs
+
+    Parameters
+    ----------
+    bids
+    item_dicoms
+    outname
+    outname_bids
+    prov_file
+    scaninfo
+    tempdirs
+    with_prov
+    min_meta
+
+    Returns
+    -------
+
+    """
+    from nipype import Node, Function
+    tmpdir = tempdirs(prefix='embedmeta')
+
+    embedfunc = Node(Function(input_names=['dcmfiles', 'niftifile', 'infofile',
+                                           'bids_info', 'force', 'min_meta'],
+                              output_names=['outfile', 'meta'],
+                              function=embed_nifti),
+                     name='embedder')
+    embedfunc.inputs.dcmfiles = item_dicoms
+    embedfunc.inputs.niftifile = op.abspath(outname)
+    embedfunc.inputs.infofile = op.abspath(scaninfo)
+    embedfunc.inputs.min_meta = min_meta
+    if bids:
+        embedfunc.inputs.bids_info = load_json(op.abspath(outname_bids))
+    else:
+        embedfunc.inputs.bids_info = None
+    embedfunc.inputs.force = True
+    embedfunc.base_dir = tmpdir
+    cwd = os.getcwd()
+    try:
+        # MG - still a bug?
```

Comment:
I think, it is safe to remove this comment.  It was due to retransmission with fixed up record.